firefox-beta-bin, firefox-devedition-bin: remove

taku0 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

I would like to remove `firefox-beta-bin` and `firefox-devedition-bin` as suggested by @mweinelt: https://github.com/NixOS/nixpkgs/pull/412533#issuecomment-2923855160

- They are updated too frequently.
- @edolstra opposed to introduce unstable packages to nixpkgs: #16860
- @garbas, the original maintainer and who introduced `firefox-beta-bin` and `firefox-devedition-bin`, reacted :+1: to my proposal here: https://github.com/NixOS/nixpkgs/pull/412533#issuecomment-2924213015

Questions:

- Does anyone object?
- What's the proper procedure to remove packages from nixpkgs?
- Should we update the packages until 25.11?


cc: @garbas @zimbatm @booxter @colemickens @Scrumplex @jopejoe1 @mweinelt

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

I have tested `firefox-bin` still works, and building `firefox-beta-bin` and `firefox-devedition-bin` results in error messages defined in `aliases.nix`.

- Built on platform(s)
  - [X] x86_64-linux (I have tested `firefox-bin`)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

taku0 committed:
firefox-beta-bin, firefox-devedition-bin: remove

mweinelt commented:
> Should we update the packages until 25.11?

One option is to make the -bin versions an alias for the source version, if those will be maintained going forward.

taku0 commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -17,6 +17,8 @@
 
 - `base16-builder` node package has been removed due to lack of upstream maintenance.
 
+- `firefox-beta-bin` and `firefox-devedition-bin` are removed. Use `firefox-beta`, ``firefox-devedition`, or `firefox-bin` instead.
```
Fixed.

taku0 commented:
> One option is to make the -bin versions an alias for the source version, if those will be maintained going forward.

- Changed `firefox-beta-bin` and `firefox-devedition-bin` to aliases to `firefox-beta` and `firefox-devedition` with warning message.
- Removed the entry from rl-2511.section.md

andersk commented on pkgs/applications/networking/browsers/firefox-bin/default.nix:
```diff
@@ -14,7 +14,6 @@
   pciutils,
   pipewire,
   adwaita-icon-theme,
-  channel,
```
What do you mean? Are you suggesting Mozilla developers should provide `firefox-beta-bin` and `firefox-devedition-bin` within nixpkgs? Didn’t you explain above why you think these don’t belong in nixpkgs? Keeping them outside nixpkgs seems to be the only realistic option.

taku0 replied:
I hope they will maintain `firefox-bin` (and `firefox-esr-bin`, if possible) within `nixpkgs`.

`firefox-bin` is a package provided by `nixpkgs` but `nixpkgs-mozilla` and `flake-firefox-nightly` are not part of `nixpkgs`. `nixpkgs` no longer requires `channel` parameter.

I believe (though I am unsure if you or other nixpkgs maintainers share this view) that `nixpkgs` has no obligation to maintain the parameter solely for the benefit of `nixpkgs-mozilla` and `flake-firefox-nightly`. I believe the function defined by `pkgs/applications/networking/browsers/firefox-bin/default.nix` is not a public API.

What's your opinion?

andersk replied:
Well, if I’m being honest, my opinion is that the amount of weekly nixpkgs breakage one has to put up with to keep a NixOS system updated is off-putting to a lot of potential users, including those very upstream developers that you’d like to see maintaining their packages in nixpkgs. Yeah, I do it, and I’m still here, and I don’t intend to disparage what volunteers do with their time. But since you asked, I’d say it’s worth thinking a bit about whether we can do better, beyond the minimum “obligation”—was it really important to remove the `channel` argument here (and to backport this breaking change to stable)?

I use Firefox Beta myself, and it sounds like you still don’t want `firefox-beta-bin` in nixpkgs, even if hypothetically Mozilla developers were to offer to maintain it.
